sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AuthenticationProvider.java

```diff
@@ -0,0 +1,170 @@
+package com.microsoft.appcenter.analytics;
+
+import com.microsoft.appcenter.utils.AppCenterLog;
+import com.microsoft.appcenter.utils.HandlerUtils;
+import com.microsoft.appcenter.utils.HashUtils;
+import com.microsoft.appcenter.utils.TicketCache;
+
+import java.util.Date;
+
+import static com.microsoft.appcenter.analytics.Analytics.LOG_TAG;
+
+public class AuthenticationProvider {
+
+    /**
+     * Ratio threshold of expiry time to refresh token a bit before it's expired.
+     */
+    private static final double REFRESH_THRESHOLD = 0.9;
+
+    /**
+     * The type.
+     */
+    private final Type mType;
+
+    /**
+     * The ticket key for this authentication provider.
```

Comment:
This would be a good place to describe what a ticket is if needed.

Reply:
No the right place to add details is the contructor. The field is private. Will add details to constuctor.